packages/design-system/src/components/IconButton/IconButton.tsx

```diff
@@ -1,45 +1,29 @@
 import * as React from 'react';
 
-import { styled } from 'styled-components';
+import { css, styled } from 'styled-components';
 
 import { PolymorphicRef, PropsToTransientProps } from '../../types';
 import { AccessibleIcon } from '../../utilities/AccessibleIcon';
 import { forwardRef } from '../../utilities/forwardRef';
-import { BaseButton, BaseButtonComponent, BaseButtonProps } from '../BaseButton';
-import { Flex, FlexComponent } from '../Flex';
+import { ButtonProps } from '../Button';
+import { getActiveStyle, getDisabledStyle, getHoverStyle, getVariantStyle } from '../Button/utils';
+import { Flex, FlexComponent, FlexProps } from '../Flex';
 import { Tooltip } from '../Tooltip';
 
-// TODO: we should align the default state in v2 with the Button
-// component
-const VARIANT_DEFAULT = 'tertiary';
-const VARIANT_SECONDARY = 'secondary';
-
-const SIZES = ['S', 'M', 'L'] as const;
-const VARIANTS = [VARIANT_DEFAULT, VARIANT_SECONDARY] as const;
-
-type IconButtonSize = (typeof SIZES)[number];
-type IconButtonVariant = (typeof VARIANTS)[number];
-
-type IconButtonProps<C extends React.ElementType = 'button'> = BaseButtonProps<C> & {
-  children: React.ReactNode;
-  /**
-   * This isn't visually rendererd, but required for accessibility.
-   */
-  label: string;
-  onClick?: React.MouseEventHandler<HTMLButtonElement>;
-  /**
-   * @default 'S'
-   */
-  size?: IconButtonSize;
-  /**
-   * @default 'tertiary'
-   */
-  variant?: IconButtonVariant;
-  /**
-   * @default true
-   */
-  withTooltip?: boolean;
-};
+type IconButtonProps<C extends React.ElementType = 'button'> = FlexProps<C> &
+  Pick<ButtonProps, 'size' | 'variant'> & {
+    children: React.ReactNode;
+    disabled?: boolean;
+    /**
+     * This isn't visually rendererd, but required for accessibility.
+     */
+    label: string;
+    onClick?: React.MouseEventHandler<HTMLButtonElement>;
+    /**
+     * @default true
+     */
+    withTooltip?: boolean;
```

Comment:
When would we not want a tooltip? It seems to me like it allows unaccessible implementations

Reply:
It's still accessible, the implementation of an IconButton wraps `AccessibleIcon` which renders an invisible label. We have many buttons in strapi that are IconButtons without a tooltip at all, you used to do it by passing `aria-label` instead of `label`.